src/components/cc-domain-management/cc-domain-management.js

```diff
@@ -106,99 +116,77 @@ export class CcDomainManagement extends LitElement {
         suggestedElement.focus();
       }
       else {
-        this._emptyMessageRef?.value?.focus();
+        this._emptyMessageRef.value?.focus();
       }
     });
 
     new LostFocusController(this, '.mark-primary', () => {
-      /** @type {HTMLDivElement|null} */
-      const primaryDomainCard = this.shadowRoot?.querySelector('.primary');
-      primaryDomainCard?.focus();
+      focusBySelector(this, '.primary');
     });
   }
 
   /**
-   * @param {string} domainWithPathPrefix
-   * @param {boolean} isHttpOnly
-   * @returns {string}
-   */
-  _getDomainUrl (domainWithPathPrefix, isHttpOnly) {
-    const escapedDomainWithPathPrefix = domainWithPathPrefix.replace('*.', 'www.');
-    return isHttpOnly ? 'http://' + escapedDomainWithPathPrefix : 'https://' + escapedDomainWithPathPrefix;
-  }
-
-  /**
-   * @param {string} domain
+   * @param {string} hostname
    * @returns {FormError|null} The error code or `null` if no error
    */
-  _getErrorCode (domain) {
-    if (domain.length === 0) {
-      return 'empty';
-    }
+  _getErrorCode (hostname) {
+    try {
+      const { pathname } = parseDomain(hostname?.trim());
 
-    if (!domain.includes('.')) {
-      return 'invalid-format';
-    }
+      if (pathname !== '/') {
+        return { code: 'hostname-contains-path', pathWithinHostname: pathname };
+      }
 
-    if (domain.includes('*') && !domain.startsWith('*.')) {
-      return 'invalid-wildcard';
+      return null;
     }
+    catch (error) {
+      if (error instanceof DomainError) {
+        return error.message;
+      }
 
-    // TODO: move to `URL.parse()` when support is good enough
-    try {
-      // With firefox, 'https://*.toto.com' is considered invalid so we strip it off for the test because we know this part is valid
-      new URL('https://' + domain.replace(/^\*\./, ''));
-    }
-    catch {
       return 'invalid-format';
     }
 
-    if (domain.match(/\/[^$]/)) {
-      return 'contains-path';
-    }
-
-    return null;
   }
 
   /**
-   * @param {FormError} errorCode
+   * @param {FormError} error
    * @returns {string|null}
    */
-  _getErrorMessage (errorCode) {
-    switch (errorCode) {
-      case 'empty':
-        return i18n('cc-domain-management.form.domain.error.empty');
-      case 'invalid-format':
-        return i18n('cc-domain-management.form.domain.error.format');
-      case 'invalid-wildcard':
-        return i18n('cc-domain-management.form.domain.error.wildcard');
-      case 'contains-path':
-        return i18n('cc-domain-management.form.domain.error.contains-path', { path: this._lastPathWithinDomainInputValue });
-      default:
-        return null;
+  _getErrorMessage (error) {
```

Comment:
This has been refactored because `error` may be an object when the error is of type `hostname-contains-path`.

The error object in this context allows us to store the `pathWithinHostname` value to show it within the error. I used to run `parseDomain` twice to get the `pathWithinHostname` value (once when checking the error and once when setting the global reactive prop to store the value).